src/main/resources/META-INF/reaktivity/socks.idl

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+scope socks
+{
+    enum SocksMode
+    {
+        FORWARD,
+        REVERSE
+    }
+
+    scope control
+    {
+        struct SocksRouteEx
+        {
+            socks::SocksMode mode;
+            string16 destAddrPort;
```

Comment:
Is this needed?  And what for?

If it is needed, I would suggest using the same address ext  (maybe the same extension) as TCP: 
```
            tcp::TcpAddress localAddress;
            uint16 localPort;
```

https://github.com/reaktivity/nukleus-tcp.spec/blob/develop/src/main/resources/META-INF/reaktivity/tcp.idl

Reply:
This is an attempt to match the 3 fields in the [Socks RFC](https://tools.ietf.org/html/rfc1928#section-4) since in case of `0x03` for `ATYP` a domain name should be sent. The initial idea was to describe the whole 3 fields (`ATYP`, `DST.ADDR`, `DST.PORT`) but went for a simpler String mapping.

Reply:
Ok with me.  Thoughts @jfallows